.github/workflows/main.yml

```diff
@@ -0,0 +1,131 @@
+name: CI
+
+on:
+  push:
+    branches:
+    - master
+  pull_request:
+    branches:
+    - master
+
+jobs:
+  lint:
+    name: Check lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+      - name: Cache pip
+        uses: actions/cache@v2
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-pip-${{ hashFiles('requirements.txt') }}
+          restore-keys: |
+            ${{ runner.os }}-pip-
+            ${{ runner.os }}-
+      - name: Install dependencies
+        run: |
+          set -xe
+          python -VV
+          python -m pip install --upgrade tox
```

Comment:
so this would upgrade tox but not potential dependencies if they're already installed (we'd need the `eager` update strategy).

Why the upgrade ? If tox is already installed, by whom, at which version ? Would there be a way to avoid installing tox altogether ? Or if we install it, just pip install because we know we're in a controlled clean env ?

Reply:
> so this would upgrade tox but not potential dependencies if they're already installed

yes

> Why the upgrade ?

No need actually

> If tox is already installed, by whom, at which version ?

The environment is supposed to be clean. `tox` should not be installed yet.

> Would there be a way to avoid installing tox altogether ?

I do not think so. How would you check linting if tox is not installed ?

> just pip install because we know we're in a controlled clean env ?

Yep, this is what I think is the best thing to do :+1:

Reply:
>> Would there be a way to avoid installing tox altogether ?
> 
> I do not think so. How would you check linting if tox is not installed ?

Nah :) I mean "if somehow tow was already present in our environment, maybe we could avoid touching it and let whoever made it available manage it" but you answered: tow is not installed and the env is clean. So let's `pip install` all the things !